Samples/Tutorials/BasicTutorial2.cpp

```diff
@@ -0,0 +1,230 @@
+
+    // register our scene with the RTSS
+    Ogre::RTShader::ShaderGenerator* shadergen = Ogre::RTShader::ShaderGenerator::getSingletonPtr();
+    shadergen->addSceneManager(scnMgr);
+
+    // -- tutorial section start --
+    //! [cameracreate]
+    Ogre::SceneNode* camNode = scnMgr->getRootSceneNode()->createChildSceneNode();
+    Ogre::Camera* cam = scnMgr->createCamera("myCam");
+    //! [cameracreate]
+
+    //! [cameraposition]
+    camNode->setPosition(200, 300, 400);
+    camNode->lookAt(Ogre::Vector3(0, 0, 0), Ogre::Node::TransformSpace::TS_WORLD);
+    //! [cameraposition]
+
+    //! [cameralaststep]
+    cam->setNearClipDistance(5);
+    camNode->attachObject(cam);
+    //! [cameralaststep]
+
+    //! [addviewport]
+    Ogre::Viewport* vp = mWindow->addViewport(cam);
+    //! [addviewport]
+
+    //! [viewportback]
+    vp->setBackgroundColour(Ogre::ColourValue(0, 0, 0));
+    //! [viewportback]
+
+    //! [cameraratio]
+    cam->setAspectRatio(
+      Ogre::Real(vp->getActualWidth()) / Ogre::Real(vp->getActualHeight())
+    );
+    //! [cameraratio]
+
+    //! [ninja]
+    Ogre::Entity* ninjaEntity = scnMgr->createEntity("ninja.mesh");
+    ninjaEntity->setCastShadows(true);
+
+    scnMgr->getRootSceneNode()->createChildSceneNode()->attachObject(ninjaEntity);
+    //! [ninja]
+
+    //! [plane]
+    Ogre::Plane plane(Ogre::Vector3::UNIT_Y, 0);
+    //! [plane]
+
+    //! [planedefine]
+    Ogre::MeshManager::getSingleton().createPlane(
+            "ground",
+            Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME,
+            plane,
+            1500, 1500, 20, 20,
+            true,
+            1, 5, 5,
+            Ogre::Vector3::UNIT_Z);
+    //! [planedefine]
+
+    //! [planecreate]
+    Ogre::Entity* groundEntity = scnMgr->createEntity("ground");
+    scnMgr->getRootSceneNode()->createChildSceneNode()->attachObject(groundEntity);
+    //! [planecreate]
+
+    //! [planenoshadow]
+    groundEntity->setCastShadows(false);
+    //! [planenoshadow]
+
+    //! [planesetmat]
+    groundEntity->setMaterialName("Examples/Rockwall");
+    //! [planesetmat]
+
+    //! [lightingsset]
+    scnMgr->setAmbientLight(Ogre::ColourValue(0, 0, 0));
+    scnMgr->setShadowTechnique(Ogre::ShadowTechnique::SHADOWTYPE_STENCIL_MODULATIVE);
+    //! [lightingsset]
+
+    //! [spotlight]
+    Ogre::Light* spotLight = scnMgr->createLight("SpotLight");
+    //! [spotlight]
+
+    //! [spotlightcolor]
+    spotLight->setDiffuseColour(0, 0, 1.0);
+    spotLight->setSpecularColour(0, 0, 1.0);
+    //! [spotlightcolor]
+
+    //! [spotlighttype]
+    spotLight->setType(Ogre::Light::LT_SPOTLIGHT);
+    //! [spotlighttype]
+
+    //! [spotlightposrot]
+    spotLight->setDirection(-1, -1, 0);
+    spotLight->setPosition(Ogre::Vector3(200, 200, 0));
+    //! [spotlightposrot]
+
+    //! [spotlightrange]
+    spotLight->setSpotlightRange(Ogre::Degree(35), Ogre::Degree(50));
```

Comment:
you can just do ˋusing namespace Ogre;ˋ somewhere instead of using the Ogre:: prefix everywhere to keep things more readable.